File: src/main/java/gg/mineral/server/network/packet/play/clientbound/EffectPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record EffectPacket(int effectId, int x, short y, int z, int data, boolean disableRelativeVolume)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(effectId);
        os.writeInt(x);
        os.writeByte(y);
        os.writeInt(z);
        os.writeInt(data);
        os.writeBoolean(disableRelativeVolume);
    }

    @Override
    public byte getId() {
        return 0x28;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EffectPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EffectPacket(
    val effectId: Int,
    val x: Int,
    val y: UByte,
    val z: Int,
    val data: Int,
    val disableRelativeVolume: Boolean
) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(effectId, x)
        os.writeByte(y.toInt())
        os.writeInt(z, data)
        os.writeBoolean(disableRelativeVolume)
    }

    override val id: Byte
        get() = 0x28
}
